src/main/java/io/cdap/plugin/azure/face/extractor/AzureFaceExtractorConfig.java

```diff
@@ -0,0 +1,163 @@
+
+/**
+ * Azure Face Extractor plugin configuration.
+ */
+public class AzureFaceExtractorConfig extends PluginConfig {
+  public static final String SOURCE_FIELD_NAME = "sourceFieldName";
+  public static final String CONTINUE_ON_ERROR = "continueOnError";
+  public static final String FACES_SUBSCRIPTION_KEY = "facesSubscriptionKey";
+  public static final String EMOTION_SUBSCRIPTION_KEY = "emotionSubscriptionKey";
+
+  @Name(SOURCE_FIELD_NAME)
+  @Description("Specifies the input field containing the binary pdf data.")
+  private final String sourceFieldName;
+
+  @Name(CONTINUE_ON_ERROR)
+  @Macro
+  @Description("Set to true if this plugin should ignore errors.")
+  private Boolean continueOnError;
+
+  @Name(FACES_SUBSCRIPTION_KEY)
+  @Description("The Azure Faces API subscription key.")
+  @Macro
+  private String facesSubscriptionKey;
+
+  @Name(EMOTION_SUBSCRIPTION_KEY)
+  @Description("The Azure emotion API subscription key.")
+  @Macro
+  private String emotionSubscriptionKey;
+
+  public AzureFaceExtractorConfig(String sourceFieldName, Boolean continueOnError,
+                String facesSubscriptionKey, String emotionSubscriptionKey) {
+    this.sourceFieldName = sourceFieldName;
+    this.continueOnError = continueOnError;
+    this.facesSubscriptionKey = facesSubscriptionKey;
+    this.emotionSubscriptionKey = emotionSubscriptionKey;
+  }
+
+  private AzureFaceExtractorConfig(Builder builder) {
+    sourceFieldName = builder.sourceFieldName;
+    continueOnError = builder.continueOnError;
+    facesSubscriptionKey = builder.facesSubscriptionKey;
+    emotionSubscriptionKey = builder.emotionSubscriptionKey;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(AzureFaceExtractorConfig copy) {
+    return builder()
+      .setSourceFieldName(copy.sourceFieldName)
+      .setContinueOnError(copy.continueOnError)
+      .setFacesSubscriptionKey(copy.facesSubscriptionKey)
+      .setEmotionSubscriptionKey(copy.emotionSubscriptionKey);
+  }
+
+  public String getSourceFieldName() {
+    return sourceFieldName;
+  }
+
+  public boolean getContinueOnError() {
+    return continueOnError;
+  }
+
+  public String getFacesSubscriptionKey() {
+    return facesSubscriptionKey;
+  }
+
+  public String getEmotionSubscriptionKey() {
+    return emotionSubscriptionKey;
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    if (inputSchema == null) {
+      failureCollector.addFailure("Could not get the input schema to validate.",
+                                  "Please check input schema.");
+      return;
+    }
+    if (inputSchema.getField(sourceFieldName) == null) {
+      failureCollector.addFailure(
+        String.format("Source field '%s' was not present in the input schema. Schema: '%s'.", sourceFieldName,
+                      inputSchema.toString()), "Provide Source Field that present in the input schema.")
```

Comment:
lets not stringify schema in the error message

Reply:
error - `Source field '%s' must be present in the input schema.`
corrective action - null

Reply:
Done.